modules/openid_connect/app/models/openid_connect/provider/hash_builder.rb

```diff
@@ -30,55 +30,93 @@
 
 module OpenIDConnect
   module Provider::HashBuilder
-    def attribute_map
-      OpenIDConnect::Provider::MAPPABLE_ATTRIBUTES
-        .index_with { |attr| public_send(:"mapping_#{attr}") }
-        .compact_blank
+    BACKCHANNEL_LOGOUT_CALLBACK = ->(logout_token) do
+      ::OpenProject::OpenIDConnect::SessionMapper.handle_logout(logout_token)
     end
 
-    def to_h # rubocop:disable Metrics/AbcSize
+    STATE_GENERATOR = -> { SecureRandom.hex(42) }
+
+    def to_omniauth_hash # rubocop:disable Metrics/AbcSize
+      client_options = {
+        identifier: client_id,
+        secret: client_secret,
+        authorization_endpoint:,
+        token_endpoint:,
+        userinfo_endpoint:,
+        jwks_uri:,
+        end_session_endpoint:,
+        host: host || ::Storages::UrlBuilder.host_from_endpoint(authorization_endpoint),
```

Comment:
Generally looks good and I remember that we talked about this. However, I think now that we use this helper class from the `openid_connect` module, we should move it into the core and not leave it in `Storages`.

I.e. (including moving the file):

```suggestion
        host: host || ::UrlBuilder.host_from_endpoint(authorization_endpoint),
```

or maybe

```suggestion
        host: host || ::OpenProject::UrlBuilder.host_from_endpoint(authorization_endpoint),
```